File: packages/python/src/randino/name/random_name_details.py
"""Generating person names with both scripts and the choices behind each one."""

from randino._types import NameDetail, NameGenderOption, NameLanguageOption
from randino.name._generator import generate_name_details


def random_name_details(
    *,
    language: NameLanguageOption = "all",
    gender: NameGenderOption = "all",
    count: int = 1,
    style: int = 0,
    min_length: int | None = None,
    max_length: int | None = None,
    include_surname: bool = True,
    include_middle_name: bool = False,
    starts_with: str = "",
    unique: bool = False,
) -> list[NameDetail]:
    """Generate person names with both scripts and the choices behind each one.

    Takes the same options as `random_name`, except `script` — every name is
    returned in its native form and romanized at the same time. Useful when the
    language is mixed (`"all"`), where `language` tells you what each name is, or
    when you want to show a name next to its English pronunciation.

    Example:
        >>> random_name_details(language="ko")
        [NameDetail(native='김민준', roman='Kim Minjun', language='ko', gender='male')]
    """
    return generate_name_details(
        language=language,
        gender=gender,
        count=count,
        style=style,
        min_length=min_length,
        max_length=max_length,
        include_surname=include_surname,
        include_middle_name=include_middle_name,
        starts_with=starts_with,
        unique=unique,
    )

File: packages/dart/lib/src/name/random_name_details.dart
import 'package:randino/src/name/name_generator.dart';
import 'package:randino/src/types.dart';

/// Generate person names with both scripts and the choices behind each one.
///
/// Takes the same parameters as `randomName` except `script` — every name is
/// returned in its native form and romanized at the same time. Useful when the
/// language is mixed, where [NameDetail.language] tells you what each name is,
/// or when you want to show a name next to its English pronunciation.
///
/// ```dart
/// randomNameDetails(language: NameLanguage.ko);
/// // [NameDetail(김민준, Kim Minjun, ko, male)]
/// ```
List<NameDetail> randomNameDetails({
  NameLanguage? language,
  NameGender? gender,
  int count = 1,
  int style = 0,
  int? minLength,
  int? maxLength,
  bool includeSurname = true,
  bool includeMiddleName = false,
  String? startsWith,
  bool unique = false,
}) => generateNameDetails(
  language: language,
  gender: gender,
  count: count,
  style: style,
  minLength: minLength,
  maxLength: maxLength,
  includeSurname: includeSurname,
  includeMiddleName: includeMiddleName,
  startsWith: startsWith,
  unique: unique,
);
